obstacle_stop_planner_refine/include/obstacle_stop_planner/one_step_polygon.hpp

```diff
@@ -0,0 +1,159 @@
+        std::cos(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        base_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width),
+        base_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        base_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width),
+        base_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width)));
+  }
+  // next step
+  {
+    double yaw = getYawFromGeometryMsgsQuaternion(next_step_pose.orientation);
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) -
+        std::sin(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) +
+        std::cos(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) -
+        std::sin(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (vehicle_info_->wheel_base_m_ + vehicle_info_->front_overhang_m_) +
+        std::cos(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (-vehicle_info_->vehicle_width_m_ / 2.0 - expand_width)));
+    one_step_move_vehicle_corner_points.push_back(
+      cv::Point2d(
+        next_step_pose.position.x + std::cos(yaw) * (-vehicle_info_->rear_overhang_m_) -
+        std::sin(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width),
+        next_step_pose.position.y + std::sin(yaw) * (-vehicle_info_->rear_overhang_m_) +
+        std::cos(yaw) * (vehicle_info_->vehicle_width_m_ / 2.0 + expand_width)));
+  }
+  polygon_ = convexHull(one_step_move_vehicle_corner_points);
+  boost_polygon_ = ConvertToBoostPolygon(polygon_);
+}
```

Comment:
Please make this function use `create_vehicle_footprint()`. It's in lane_departure_checker now. You can copy it to this node tentatively, and finally, we should consider putting it into autoware_utils.

Reply:
@kenji-miyake Thanks. I copied `create_vehicle_footprint()` and use it.
https://github.com/tier4/obstacle_stop_planner_refine/pull/3/files#diff-dc8dc3b37a8753355ca4741003d69268a20628ce62be0068c4c4876c1b5a4f4aR21

Reply:
Wow, so nice!